trieste/acquisition/batch.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2022 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import functools
+
+import tensorflow as tf
+
+from ..types import TensorType
+from .interface import AcquisitionFunction
+
+
+def batchify_acquisition_function(
+    fn: AcquisitionFunction,
+    batch_size: int,
+) -> AcquisitionFunction:
+    """
+    A wrapper around an :const:`AcquisitionFunction` to split its input into batches.
+    Splits `x` into batches along the first dimension, calls `fn` on each batch, and then stitches
+    the results back together, so that it looks like `fn` was called with all of `x` in one batch.
+    :param fn: Acquisition function to call with batches of data.
+    :param batch_size: Call fn with tensors of at most this size.
+    :returns Batched acquisition function.
+    """
+    if batch_size <= 0:
+        raise ValueError(f"batch_size must be positive, got {batch_size}")
+
+    @functools.wraps(fn)
+    def wrapper(x: TensorType) -> TensorType:
+        x = tf.convert_to_tensor(x)
+
+        # this currently assumes leading dimension of x is batch dimension.
```

Comment:
again, batch dimension is what we call the -2 dimension in Trieste. This naming is a bit confusing